internal/transport/http2_client.go

```diff
@@ -1232,18 +1232,27 @@ func (t *http2Client) handleGoAway(f *http2.GoAwayFrame) {
 	if upperLimit == 0 { // This is the first GoAway Frame.
 		upperLimit = math.MaxUint32 // Kill all streams after the GoAway ID.
 	}
+
+	t.prevGoAwayID = id
+	if len(t.activeStreams) == 0 {
+		t.mu.Unlock()
+		t.Close(connectionErrorf(true, nil, "received goaway and there are no active streams"))
+		return
+	}
+
+	streamsToClose := make([]*Stream, 0)
```

Comment:
`var streamsToClose []*Stream`

Or delete the `,0` but `make` is unnecessary for a slice either way if you aren't allocating anything.

Reply:
Switched. You initially suggested make for the capacity length. Why was that?

Reply:
If you specify the capacity then it won't need to do reallocations as you add to it.  In this case, since we don't have any idea how many active streams are before/after the ID (and it isn't worth computing it first) then this should be fine.  It will need reallocations (Go doubles the slice capacity as you add so it's O(logN)), but this is a rare case and not worth optimizing.